packages/graphql/tests/integration/rfcs/global-authentication-jwks-plugin.int.test.ts

```diff
@@ -0,0 +1,210 @@
+            )
+        ).toBeTruthy();
+        expect(gqlResult.data).toBeNull();
+    });
+
+    test("should fail if invalid JWT token is provided and global authentication is enabled", async () => {
+        const neoSchema = new Neo4jGraphQL({
+            typeDefs,
+            plugins: {
+                auth: new Neo4jGraphQLAuthJWKSPlugin({
+                    jwksEndpoint: "https://myAuthTest.auth0.com/.well-known/jwks.json",
+                    globalAuthentication: true,
+                }),
+            },
+        });
+
+        const gqlResult = await graphql({
+            schema: await neoSchema.getSchema(),
+            source: query,
+            contextValue: neo4j.getContextValues({
+                request: {
+                    headers: { Authorization: `Bearer xxx.invalidtoken.xxxx` },
+                },
+            }),
+        });
+
+        expect(gqlResult.errors).toBeDefined();
+        expect(
+            (gqlResult.errors as unknown as Neo4jGraphQLAuthenticationError[]).some((el) =>
+                el.message.includes("Enabled global authentication requires a valid JWT token")
+            )
+        ).toBeTruthy();
+        expect(gqlResult.data).toBeNull();
+    });
+
+    test("should not fail if valid JWT token is present and global authentication is enabled", async () => {
+        const neoSchema = new Neo4jGraphQL({
+            typeDefs,
+            plugins: {
+                auth: new Neo4jGraphQLAuthJWKSPlugin({
+                    jwksEndpoint: "https://myAuthTest.auth0.com/.well-known/jwks.json",
+                    globalAuthentication: true,
+                }),
+            },
+        });
+
+        // Start the JWKS Mock Server Application
+        jwksMock.start();
+
+        const accessToken = jwksMock.token({
+            iat: 1600000000,
+        });
+
+        const gqlResult = await graphql({
+            schema: await neoSchema.getSchema(),
+            source: query,
+            contextValue: neo4j.getContextValues({
+                request: {
+                    headers: { Authorization: `Bearer ${accessToken}` },
+                },
+            }),
+        });
+
+        expect(gqlResult.errors).toBeUndefined();
+        expect((gqlResult.data as any)[testMovie.plural]).toHaveLength(0);
+    });
+});
+
+const createContext = () => {
+    // This creates the local PKI
+    const jwksMock = createJWKSMock("https://myAuthTest.auth0.com");
+
+    // We start our app.
+    const server = createApp({
+        jwksUri: "https://myAuthTest.auth0.com/.well-known/jwks.json",
+    }).listen();
+
+    const request = supertest(server);
+    return {
+        jwksMock,
+        request,
+        server,
+    };
+};
+
+const tearDown = async ({ jwksMock, server }) => {
+    await server.close();
+    await jwksMock.stop();
+};
+
+const createApp = ({ jwksUri }) => {
+    const app = new Koa();
```

Comment:
Are we using a full Koa server for this test?
Maybe this should be an `e2e` test instead?

A testing apollo server already exists for this kind of test with `supertest` you could use that or move this koa setup to a separate setup file there.

Reply:
I'll try to turn this into an `e2e` test instead.